server/manager/src/main/java/org/apache/accumulo/manager/tableOps/bulkVer2/LoadFiles.java

```diff
@@ -157,62 +160,149 @@ void start(Path bulkDir, Manager manager, long tid, boolean setTime) throws Exce
 
   private static class OnlineLoader extends Loader {
 
+    private final int maxConnections;
     long timeInMillis;
     String fmtTid;
     int locationLess = 0;
 
-    // track how many tablets were sent load messages per tablet server
-    MapCounter<HostAndPort> loadMsgs;
+    int tabletsAdded;
 
     // Each RPC to a tablet server needs to check in zookeeper to see if the transaction is still
     // active. The purpose of this map is to group load request by tablet servers inorder to do less
     // RPCs. Less RPCs will result in less calls to Zookeeper.
     Map<HostAndPort,Map<TKeyExtent,Map<String,MapFileInfo>>> loadQueue;
     private int queuedDataSize = 0;
 
+    public OnlineLoader(AccumuloConfiguration configuration) {
+      super();
+      this.maxConnections = configuration.getCount(Property.MANAGER_BULK_MAX_CONNECTIONS);
+    }
+
     @Override
     void start(Path bulkDir, Manager manager, long tid, boolean setTime) throws Exception {
       super.start(bulkDir, manager, tid, setTime);
 
       timeInMillis = manager.getConfiguration().getTimeInMillis(Property.MANAGER_BULK_TIMEOUT);
       fmtTid = FateTxId.formatTid(tid);
 
-      loadMsgs = new MapCounter<>();
+      tabletsAdded = 0;
 
       loadQueue = new HashMap<>();
     }
 
+    private static class Client {
+      final HostAndPort server;
+      final TabletClientService.Client service;
+
+      private Client(HostAndPort server, TabletClientService.Client service) {
+        this.server = server;
+        this.service = service;
+      }
+    }
+
     private void sendQueued(int threshhold) {
       if (queuedDataSize > threshhold || threshhold == 0) {
-        loadQueue.forEach((server, tabletFiles) -> {
+        var sendTimer = Timer.startNew();
+
+        List<Client> clients = new ArrayList<>();
+        try {
+
+          // Send load messages to tablet servers spinning up work, but do not wait on results.
+          loadQueue.forEach((server, tabletFiles) -> {
+
+            if (log.isTraceEnabled()) {
+              log.trace("{} asking {} to bulk import {} files for {} tablets", fmtTid, server,
+                  tabletFiles.values().stream().mapToInt(Map::size).sum(), tabletFiles.size());
+            }
+
+            // On the server side tablets are processed serially with a write to the metadata table
+            // done for each tablet. Chunk the work up for a tablet server up so that it can be sent
+            // over multiple connections allowing it to run in parallel on the server side. This
+            // allows multiple threads on a single tserver to do metadata writes for this bulk
+            // import.
```

Comment:
```suggestion
            // Tablet servers process tablets serially and perform a single metadata table write for each tablet. 
            // Break the work into per-tablet chunks so it can be sent over multiple connections 
            // to the tserver, allowing each chunk to be run in parallel on the server side.
            // This allows multiple threads on a single tserver to do metadata writes for this bulk
            // import.
```